transacoes/src/main/java/com/amanda/transacoes/services/relatorios/InformacoesDeClienteService.java

```diff
@@ -0,0 +1,207 @@
+import com.amanda.transacoes.enums.TipoOperacaoEnum;
+import com.amanda.transacoes.models.ClienteModel;
+import com.amanda.transacoes.models.TransacaoModel;
+import com.amanda.transacoes.repositories.ClienteRepository;
+import com.amanda.transacoes.repositories.TransacaoRepository;
+import java.util.stream.Stream;
+
+@Service
+public class InformacoesDeClienteService {
+
+    @Autowired
+    private ClienteRepository clienteRepository;
+
+    @Autowired
+    private TransacaoRepository transacaoRepository;
+
+    public Map<String, Double> getSaldoBanco() {
+        
+        double saldoBanco = clienteRepository.findAll()
+            .stream()
+            .mapToDouble(ClienteModel::getSaldo)
+            .sum();
+
+        Map<String, Double> relatorio = new  HashMap<>();
+
+        relatorio.put("Saldo do banco", saldoBanco);
+
+        return relatorio;
+    }
+
+    public Map<String, Integer> getQuantidadeClientesAtivos() {
+
+        int clientesAtivos = (int) clienteRepository.findAll()   
+            .stream()
+            .filter(ClienteModel::getAtivo)
+            .count();
+
+        Map<String, Integer> relatorio = new  HashMap<>();
+
+        relatorio.put("Quantidade de clientes ativos", clientesAtivos);
+
+        return relatorio;
+    }
+
+    public List<ClienteETiposOperacaoDto> getQuantidadeDeTipoOperacaoPorCliente() {
+        List<TransacaoModel> transacoes = transacaoRepository.findAll();
+
+        List<ClienteETiposOperacaoDto> listaClientes =  listarTransacoesPorConta(transacoes).entrySet().stream()
+            .map(entry -> {
+                String ccCliente = entry.getKey();
+                List<TransacaoModel> transacoesPorConta = entry.getValue();
+
+                ClienteModel cliente = clienteRepository.findByNumConta(ccCliente)
+                    .orElse(null);
+
+                ClienteETiposOperacaoDto clienteDto = new ClienteETiposOperacaoDto(cliente);
+
+                transacoesPorConta.forEach(transacao -> clienteDto.incrementarTipoOperacao(transacao.getTipoOperacao()));
+
+                return clienteDto;
+            })
+            .collect(Collectors.toList()); 
+
+        return listaClientes;
+    }
+
+
+    public Map<String, List<TransacaoModel>> listarTransacoesPorConta(List<TransacaoModel> transacoes) {
+        Map<String, List<TransacaoModel>> transacoesPorCliente = new HashMap<>();
+
+        transacoesPorCliente = transacoes.stream()
+            .flatMap(transacao -> Stream.of(
+                transacao.getCcOrigem() == null || transacao.getCcOrigem().isEmpty() ? null 
+                    : Map.entry(transacao.getCcOrigem(), transacao),
+                transacao.getCcDestino() == null || transacao.getCcDestino().isEmpty() ? null 
+                    : Map.entry(transacao.getCcDestino(), transacao)
+            ))
+            .filter(Objects::nonNull)                                                               // essa funcao ja remove entradas nulas 
```

Comment:
acredito que seja melhor colocar o `filter(Objects:nonNull)` primeiro, depois outro filter pra ignorar as transacoes que `CC origem/estino` seja null. Na sequencia voce faz o `flatMap` com a certeza que tudo que está alí é valido
```suggestion
            .flatMap(transacao -> Stream.of(
                transacao.getCcOrigem() == null || transacao.getCcOrigem().isEmpty() ? null 
                    : Map.entry(transacao.getCcOrigem(), transacao),
                transacao.getCcDestino() == null || transacao.getCcDestino().isEmpty() ? null 
                    : Map.entry(transacao.getCcDestino(), transacao)
            ))
            .filter(Objects::nonNull)                                                               // essa funcao ja remove entradas nulas 
```

Reply:
Eu acabei mudando, ao invés de null eu coloquei stream.empty(), daí não precisa mais do filter(Objects:nonNull)